testing/test_driver.py

```diff
@@ -180,3 +183,60 @@ def test_pass(driver_kwargs):
     """
     )
     testdir.quick_qa("--driver", "Firefox", file_test, passed=1)
+
+
+def test_driver_retry_pass(testdir, mocker):
+    mocker.patch.dict(
+        pytest_selenium.SUPPORTED_DRIVERS,
+        CaseInsensitiveDict({"Firefox": mocker.MagicMock()}),
+    )
+    mock_retrying = mocker.spy(pytest_selenium.pytest_selenium, "Retrying")
+
+    file_test = testdir.makepyfile(
+        """
+        import pytest
+
+        @pytest.mark.nondestructive
+        def test_pass(driver):
+            assert driver
+    """
+    )
+
+    testdir.quick_qa("--driver", "Firefox", file_test, passed=1)
+    assert mock_retrying.spy_return.statistics["attempt_number"] == 1
+
+
+@pytest.mark.parametrize("max_num_attempts", ["default", 2])
+def test_driver_retry_fail(testdir, mocker, max_num_attempts):
+    mocker.patch(
+        "pytest_selenium.webdriver.firefox.webdriver.FirefoxRemoteConnection",
+        side_effect=Exception("Connection Error"),
+    )
+    mocker.patch("pytest_selenium.webdriver.firefox.webdriver.Service")
+    mocker.patch("pytest_selenium.pytest_selenium.wait_exponential", return_value=0)
+    mock_retrying = mocker.spy(pytest_selenium.pytest_selenium, "Retrying")
+
+    if max_num_attempts != "default":
+        testdir.makeini(
+            f"""
+            [pytest]
+            max_driver_init_attempts = {max_num_attempts}
+            """
+        )
+
+    file_test = testdir.makepyfile(
+        """
+        import pytest
+
+        @pytest.mark.nondestructive
+        def test_pass(driver):
+            assert True  # catch if this starts to pass
+        """
+    )
+
+    testdir.quick_qa("--driver", "Firefox", file_test, failed=1)
+    assert (
```

Comment:
Minor: this was a bit hard to parse because it ended up being formatted by black, perhaps it could be extracted to a local variable:

```python
expected_attempts = max_num_attempts if max_num_attempts != "default" else 3
assert mock_retrying.spy_return.statistics["attempt_number"] == expected_attempts
```

Reply:
Good idea!

I refined it a little bit, hope you don't mind. 😊 @nicoddemus